framework/test/driver/ServiceRegistryTest.cpp

```diff
@@ -88,6 +89,48 @@ void TestUnregisterFix(BundleContext context)
   US_TEST_CONDITION_REQUIRED(!reference.IsConvertibleTo("IBooService"), "Testing for IsConvertibleTo returning false");
 }
 
+void TestServiceReferenceMiscMembers(BundleContext context)
+{
+	struct TestServiceA : public ITestServiceA
+	{
+	};
+
+	auto s1 = std::make_shared<TestServiceA>();
+	ServiceProperties props;
+	props["string"] = std::string("A std::string");
```

Comment:
For a second I was wondering why the type is being stored in the properties. Simple names such as "foo", "bar" make it easier to read.